docs/content/src/learn/guides/integration-patterns/content-based-routing/README.md

```diff
@@ -0,0 +1,161 @@
+
+## What you'll build
+
+We shall have a service ‘stockQuote’ which accepts an http request from a client. The client appends as a query parameter in the request, the company name of which he wants to know the price of the stock. The stockQuote service identifies the company, routes the request to the relevant company’s stock quote service, obtain the response from the company’s service and returns the response to the client. If the company name is not available in the request, the service will simply respond with a 400 - Bad Request.
+
+![cbr](../../../../../../assets/img/cbr.png)
+
+## Prerequisites
+
+- [Ballerina Distribution](https://ballerina.io/learn/getting-started/)
+- A Text Editor or an IDE 
+> **Tip**: For a better development experience, install the Ballerina IDE plugin for [VS Code](https://marketplace.visualstudio.com/items?itemName=ballerina.ballerina)
+	
+## Implementation
+
+* Create a new ballerina project named ‘content-based-routing’.
+
+```bash
+$ ballerina new content-based-routing
+```
+
+* Navigate to the directory content-based-routing.
+
+* Add a new module named ‘stockquote_service’ to the project.
+
+```bash
+$ ballerina add stockquote_service
+```
+
+* Open the project with VSCode. The project structure will be like the following.
+
+```shell
+.
+├── Ballerina.toml
+└── src
+    └── stockquote_service
+        ├── main.bal
+        ├── Module.md
+        ├── resources
+        └── tests
+            ├── main_test.bal
+            └── resources
+```
+
+We can remove the file `main_test.bal` at the moment, since we're not writing any tests for our service.
+
+First let's create the services that we shall be using as backend endpoints. 
+
+* Create a new file named `ibmService.bal` file under 'stockquote_service' with the following content.
+
+```ballerina
+import ballerina/http;
+
+@http:ServiceConfig {
+    basePath: "/ibm"
+}
+service ibm on new http:Listener(8081) {
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "quote"
+    }
+    resource function quote(http:Caller caller, http:Request request) {
+        http:Response response = new;
+        response.setTextPayload("10000.00");
+        error? respond = caller->respond(response);
+    }
+}
+```
+
+This is simply a service that will run on port 8081 responding a text payload `10000.00`.
+
+* Likewise, let's create another file `msService.bal` with the following content.
+
+```ballerina
+import ballerina/http;
+
+@http:ServiceConfig {
+    basePath: "/ms"
+}
+service ms on new http:Listener(8082) {
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "quote"
+    }
+    resource function quote(http:Caller caller, http:Request request) {
+        http:Response response = new;
+        response.setTextPayload("12000.00");
+        error? respond = caller->respond(response);
+    }
+}
+```
+
+* Now let's open the main.bal file and add the following content. This is going to be our integration logic.
+
+```ballerina
```

Comment:
Normally we are inserting the code from the file itself. @pramodya1994 can provide some instruction on how.

Reply:
done